### ISSUE TITLE
http-json cursor is not persistent in registry

### ISSUE THREAD

**Issue by mahkorayem** [elastic/beats#24800]
### **Filebeat http-json info** 

- Version: 7.12
- Redhat 7 and Ubuntu 18.04
- Discussion URL: https://discuss.elastic.co/t/google-workspace-poll-loads-of-data/268255/2

### Issue:
The cursor is not saved to the registry

e.g.
```
{"op":"set","id":3}
{"k":"httpjson::https://www.googleapis.com/admin/reports/v1/activity/users/USER_AT_DOMAIN/applications/login","v":{"cursor":{},"ttl":1800000000000,"updated":[516114855620,1616949800]}}

```

### Steps:
Add this configuration in the filebeat.yml
Actually this was part of the module google_workspace and I was facing an issue so I separated this part to test it.

```
filebeat.inputs:
- type: httpjson
  config_version: 2
  interval: 10s
  auth.oauth2.provider: google
  auth.oauth2.google.jwt_file: credentials.json
  auth.oauth2.google.delegated_account: "USER_AT_DOMAIN"
  auth.oauth2.scopes:
    - https://www.googleapis.com/auth/admin.reports.audit.readonly
  request.url: https://www.googleapis.com/admin/reports/v1/activity/users/USER_AT_DOMAIN/applications/login
  request.timeout: 30s
  request.transforms:
    - set:
        target: url.params.startTime
        value: '[[.cursor.last_execution_datetime]]'
        default: '[[formatDate (now (parseDuration "-150h"))]]'
  response.split:
    target: body.items
    split:
      target: body.events
      keep_parent: true
  request.pagination:
    - set:
        target: url.params.pageToken
        value: "[[.last_response.body.nextPageToken]]"
  cursor:
    last_execution_datetime:
      value: '[[formatDate (now)]]'
  processors:
    - decode_json_fields:
        fields: ["message"]
        target: "json"
```

Team: filebeat

**Comment by mahkorayem** [comment]
Team: filebeat

**Comment by marc-gr** [comment]
Hello! The cursor will be saved in the second iteration, so all should be fine as far as the module is running for at least 2 iterations. Even though, this is fixed already as part of this PR, so after it is released cursor will be updated every time an event is published instead of once per interval. https://github.com/elastic/beats/pull/24784

**Comment by theherodied** [comment]
Ran into the same issue. I let the module run over night and ended up pulling 60gb of only login logs. Resulting in the same events tens of thousands of times.

**Comment by andrewkroh** [comment]
According to @marc-gr this will be fixed when the 7.12.1 release comes out.
